provider/core/src/marker.rs

```diff
@@ -307,61 +307,63 @@ impl AsULE for DataMarkerPathHash {
 }
 
 // Safe since the ULE type is `self`.
-unsafe impl EqULE for DataMarkerPathHash {}
+unsafe impl EqULE for DataMarkerIdHash {}
 
 /// The string path of a data marker. For example, "foo@1"
 ///
 /// ```
-/// # use icu_provider::marker::DataMarkerPath;
-/// const K: DataMarkerPath =
-///     icu_provider::marker::data_marker_path!("foo/bar@1");
+/// # use icu_provider::marker::DataMarkerId;
+/// const K: DataMarkerId =
+///     icu_provider::marker::data_marker_id!("foo/bar@1");
 /// ```
 ///
 /// The human-readable path string ends with `@` followed by one or more digits (the version
 /// number). Paths do not contain characters other than ASCII letters and digits, `_`, `/`.
 ///
-/// Invalid paths are compile-time errors (as [`data_marker_path!`](crate::marker::data_marker_path) uses `const`).
+/// Invalid paths are compile-time errors (as [`data_marker_id!`](crate::marker::data_marker_id) uses `const`).
 ///
 /// ```compile_fail,E0080
-/// # use icu_provider::marker::DataMarkerPath;
-/// const K: DataMarkerPath = icu_provider::marker::data_marker_path!("foo/../bar@1");
+/// # use icu_provider::marker::DataMarkerId;
+/// const K: DataMarkerId = icu_provider::marker::data_marker_id!("foo/../bar@1");
 /// ```
 #[derive(Debug, Copy, Clone, Eq)]
-pub struct DataMarkerPath {
+pub struct DataMarkerId {
+    /// The human-readable path string ends with `@` followed by one or more digits (the version
+    /// number). Paths do not contain characters other than ASCII letters and digits, `_`, `/`.
     #[cfg(any(feature = "export", debug_assertions))]
     debug: &'static str,
     hash: [u8; 8],
```

Comment:
Thought: `[u8; 8]` and `&'static [u8; 8]` are the same size on x64/aarch64. Maybe see what happens if you change it?

Reply:
We cannot put the hash in a `&'static` because you cannot put a computed (even const computed) value in a `&'static`. The string tagging wasn't computation, it was done with macros.

Reply:
But you can put a const-computed value in a `&'static`?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ec16d506a4d2e76ba7a17853adf32e04

Reply:
(this can be experimented-with later)